app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt

```diff
@@ -134,28 +147,45 @@ object ThreadController {
         }
     }
 
-    // TODO: Replace this with a RealmList sub query (blocked by https://github.com/realm/realm-kotlin/issues/1037)
+    // TODO: Replace this with a RealmList sub query.
     fun getThreadUnseenMessagesUids(thread: Thread): List<String> {
-        return mutableListOf<String>().apply {
-            thread.messages.forEach { message ->
-                if (!message.seen) {
-                    add(message.uid)
-                    addAll(thread.getMessageDuplicatesUids(message.messageId))
-                }
-            }
-        }
+        return getThreadMessagesAndDuplicatesUids(thread) { message -> !message.seen }
+    }
+
+    // TODO: Replace this with a RealmList sub query.
+    fun getThreadUnseenMessages(thread: Thread): List<Message> {
+        return getThreadMessagesAndDuplicates(thread) { message -> !message.seen }
     }
 
     fun getSameFolderThreadMessagesUids(thread: Thread): List<String> {
         return getThreadMessagesAndDuplicatesUids(thread) { message -> message.folderId == thread.folderId }
     }
 
+    fun getThreadFavoritesMessages(thread: Thread): List<Message> {
+        return getThreadMessagesAndDuplicates(thread) { message -> message.isFavorite && !message.isDraft }
+    }
+
     fun getThreadFavoritesMessagesUids(thread: Thread): List<String> {
         return getThreadMessagesAndDuplicatesUids(thread) { message -> message.isFavorite && !message.isDraft }
     }
```

Comment:
Is it really useful to have both, when the other is not even used ?

Reply:
I removed the `...Uids()` versions.

This refactor is in another PR: https://github.com/Infomaniak/android-mail/pull/470